plugin/pkg/admission/webhook/serviceresolver_test.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package webhook checks a webhook for configured operation admission
+package webhook
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestDefaultServiceResolver(t *testing.T) {
+	scenarios := []struct {
+		serviceName      string
+		serviceNamespace string
+		expectedOutput   string
+		expectError      bool
+	}{
+		// scenario 1: a service name along with a namespace resolves
+		{serviceName: "ross", serviceNamespace: "andromeda", expectedOutput: "https://ross.andromeda.svc"},
+		// scenario 2: a service name without a namespace resolves
+		{serviceName: "ross", expectedOutput: "https://ross"},
```

Comment:
This case should fail.  Services can't exist without a namespace.

Reply:
if a namespace is not provided it will resolve to default one i.e. ross.default
I assume this is not something we want.

Reply:
> if a namespace is not provided it will resolve to default one i.e. ross.default
> I assume this is not something we want.

Correct. We do not want this to happen.  It would be unexpected and we can't reasonable expect certificates to be signed for that kind of name.